docs/site/HasMany-relation.md

```diff
@@ -0,0 +1,161 @@
+for the above example is as follows:
+
+```ts
+// inside Customer model class
+@hasMany(Order, {keyTo: 'custId'}) orders: Order[];
+```
+
+## Configuring a HasMany relation
+
+The configuration and resolution of a `HasMany` relation takes place at the
+repository level. Once `HasMany` relation is defined on the source model, then
+there are a couple of steps involved to configure it and use it. On the source
+repository, the following are required:
+
+- Use [Dependency Injection](Dependency-injection.md) to inject an instance of
+  the target repository in the constructor of your source repository class.
+- Declare a property with the factory function type
+  `HasManyRepositoryFactory<TargetModel, typeof TargetModel.prototype.id>` on
+  the source repository class.
+- call the `_createHasManyRepositoryFactoryFor` function in the constructor of
+  the source repository class with the relation name (decorated relation propert
+  on the source model) and target repository instance and assign it the property
+  mentioned above.
+
+The following code snippet shows how it would look like:
+
+{% include code-caption.html
+content="/src/repositories/customer.repository.ts.ts" %}
+
+```ts
+import {Order, Customer} from '../models';
+import {OrderRepository} from './order.repository.ts';
+import {
+  DefaultCrudRepository,
+  juggler,
+  HasManyRepositoryFactory,
+} from '@loopback/repository';
+
+class CustomerRepository extends DefaultCrudRepository<
+  Customer,
+  typeof Customer.prototype.id
+> {
+  public orders: HasManyRepositoryFactory<Order, typeof Customer.prototype.id>;
+  constructor(
+    @inject('datasources.db') protected db: juggler.DataSource,
+    @repository(OrderRepository) orderRepository: OrderRepository,
+  ) {
+    super(Customer, db);
+    this.orders = this._createHasManyRepositoryFactoryFor(
+      'orders',
+      orderRepository,
+    );
+  }
+}
+```
+
+The following CRUD APIs are now available in the constrained target repository
+factory `orders` for instances of `customerRepository`:
+
+- `create` for creating a target model instance
+- `find` finding target model instance(s)
+- `delete` for deleting target model instance(s)
+- `patch` for patching target model instance(s)
+
+## Using HasMany constrained repository in a controller
+
+The same pattern used for ordinary repositories to expose their CRUD APIs in
+controller methods is employed for `HasMany` repositories. Once the hasMany
+relation has been defined and configured, controller methods can call the
+underlying constrained repository CRUD APIs and expose them as routes once
+decorated with
+[Route decorators](Routes.md#using-route-decorators-with-controller-methods). It
+will require the value of the foreign key and, depending on the request method,
+a partial value for the target model instance as demonstrated below.
+
+{% include code-caption.html
+content="src/controllers/customer-orders.controller.ts" %}
+
```

Comment:
Please add a short paragraph explaining that:
 - in LB3, methods for accessing related models were part of the same source model class
 - in LB4, we recommend to create a new controller for each relation in order to keep the controller classes smaller. I.e. don't add order-related methods to `CustomerController`, but create a new `CustomerOrdersController` class for them.